### PR TITLE
Feature/mongoose 5.x

### PR DESCRIPTION
**Pull request by JeffLabonte**
Update the package in package.json, Add newer version of node in travis,yml. 

Test on local machine passes.

### CODE REVIEW

**Comment by JeffLabonte** [comment]
mongoose v5 shows a lot more error. I will try to fix them! In a future pr.

File: .travis.yml [hunk 1]
```diff
@@ -5,7 +5,9 @@ node_js:
   - "0.12"
   - "0.11"
   - "0.10"
-  - "4.0"
+  - "4.0",
```
Comment (SamVerschueren) [inline .travis.yml]: Only test on `10`, `8` and `6`. Drop all the others as I'm not interested in them anymore, they are all EOL.

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,4 @@
 node_modules/
 coverage/
+*.lock
```
Comment (SamVerschueren) [inline .gitignore]: Is this for yarn?

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,4 @@
...
+.idea/
```
Comment (SamVerschueren) [inline .gitignore]: Sorry, not interested in IDE specific ignores.

**Review by SamVerschueren — CHANGES_REQUESTED** [review]
Added some comments. Let's fix the mongoose@5 issues in this PR as well, otherwise we have a broken commit.

File: .travis.yml (hunk 1 above)
Comment (JeffLabonte) [reply .travis.yml]: Sounds like a plan... will do.

File: .gitignore (hunk 2 above)
Comment (JeffLabonte) [reply .gitignore]: yes, I just noticed that npm's lock file is package-lock.json.

File: .gitignore (hunk 3 above)
Comment (JeffLabonte) [reply .gitignore]: I believe that it makes it simpler when you make commit to make sure that you aren't adding useless metadata or IDE specific that can cause issues.

File: .gitignore (hunk 2 above)
Comment (JeffLabonte) [reply .gitignore]: I removed it. I read it is useful [from this blog](https://jpospisil.com/2017/06/02/understanding-lock-files-in-npm-5.html)

**Comment by JeffLabonte** [comment]
Would it be possible to make this PR run a test. You should only see warnings of deprecated methods. The build should pass.

**Comment by JeffLabonte** [comment]
@SamVerschueren It should be all good to go!

**Comment by JeffLabonte** [comment]
@SamVerschueren When I will have a moment, I am thinking about doing a total makeover of the project. The way it has been built isn't relevant anymore. It could be done using 'async & await' or to keep the essence of the project, use the Promise native library. 

In the current state, it is difficult to maintain.

**Comment by dtzar** [comment]
@SamVerschueren @JeffLabonte Might I suggest we just merge this PR to allow compatibility with Mongoose 5.x (bump major version on the npm package) and then do a separate PR with an enhancement / "makeover" to use the promise native library (and bump another major version of the npm package)?  

We are [using this fork ](https://github.com/pibi/mongoose-seeder) which simply bumps the version number and works fine.

**Comment by JeffLabonte** [comment]
@dtzar indeed! I wanted to do so! but I didn't have time in the last months!

**Comment by dtzar** [comment]
huh, why close the PR without the fix?  Abandoning this repo officially?

**Pull request closed without merging**